fit_multiband.py

```diff
@@ -0,0 +1,255 @@
+
+        Parameters
+        ----------
+        schema : `lsst.afw.table.Schema`
+            The input schema for the reference source catalog, used to initialize the output schema.
+        kwargs
+            Additional arguments to be passed to the subclass constructor(s).
+
+        Notes
+        -----
+        This method must initialize an attribute named schema with the schema of the output catalog.
+        """
+        raise RuntimeError("Not implemented")
+
+    def run(
+        self, catexps: Iterable[CatalogExposure], cat_ref: afwTable.SourceCatalog, **kwargs
+    ) -> pipeBase.Struct:
+        """ Fit sources from a reference catalog using data from multiple exposures in the same region (patch).
+
+        Parameters
+        ----------
+        catexps : `typing.List [CatalogExposure]`
+            A list of catalog-exposure pairs in a given band. Subclasses may require there to only be one catexp per
+            band, and/or for the catalogs to contain HeavyFootprints with deblended images.
+        cat_ref : `lsst.afw.table.SourceCatalog`
+            A reference source catalog to fit. Subclasses may be configured fit only a subset of these sources.
+        kwargs
+            Additional keyword arguments.
+
+        Returns
+        -------
+        retStruct : `lsst.pipe.base.Struct`
+            A struct with a cat_output attribute containing the output measurement catalog.
+        """
+        raise RuntimeError("Not implemented")
+
+
+class MultibandFitConfig(
+    pipeBase.PipelineTaskConfig,
+    pipelineConnections=MultibandFitConnections,
+):
+    fit_multiband = pexConfig.ConfigurableField(
+        target=MultibandFitSubTask,
+        doc="Task to fit sources using multiple bands",
+    )
+
+    def get_band_sets(self):
+        dict_fit = self.toDict()['fit_multiband']
+        bands_to_fit = dict_fit.get('bands_fit')
+        if bands_to_fit is None:
+            raise RuntimeError(f'{__class__}.fit_multiband must have bands_fit attribute '
+                               f'(self.toDict()["fit_multiband"]={dict_fit})')
+        return set(bands_to_fit), set(dict_fit.get('bands_read', ()))
+
+
+class MultibandFitTask(pipeBase.PipelineTask):
+    ConfigClass = MultibandFitConfig
+    _DefaultName = "multibandFit"
+
+    def __init__(self, initInputs, **kwargs):
+        super().__init__(initInputs=initInputs, **kwargs)
+        self.makeSubtask("fit_multiband", schema=initInputs["cat_ref_schema"].schema)
+        self.cat_output_schema = afwTable.SourceCatalog(self.fit_multiband.schema)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        input_objs = [inputRefs.cats_meas, inputRefs.coadds]
+        dataIds_cats, dataIds = [[dRef.dataId for dRef in obj] for obj in input_objs]
+        if not all((x == y for x, y in zip(dataIds, dataIds_cats))):
+            raise RuntimeError(f'Mismatched cats_meas, coadds dataIds {dataIds_cats} {dataIds}')
```

Comment:
I don't think consistent ordering is guaranteed by the caller here.  Better to use dicts and reorder defensively.

If you want to check that the same data IDs are present for both dataset types, I think you can do that in `adjustQuantum`, maybe?